### ISSUE TITLE
Estimate `burrito.estimate_elbo` terminates with an error

### ISSUE THREAD

**Issue by junseonghwan** [phylovi/libsbn#185]
Script to reproduce the problem. It may need to be run multiple times before encountering the error.

```python
import libsbn
import vip.burrito
import pandas as pd

data_sets = {
    "ds7": "../ds-experiments/DS7/DS7_out.t"
}

fasta_paths = {
    "ds7": "../ds-experiments/DS7/DS7.fasta"
}

data_set = "ds7"
burn_in_fraction = 0
num_particles = 1000
branch_model_name="split"
scalar_model_name="lognormal"
optimizer_name="simple"
phylo_model_specification = libsbn.PhyloModelSpecification(
    substitution="JC69", site="constant", clock="strict"
)

# Create Burrito instance
burro = vip.burrito.Burrito(
    mcmc_nexus_path=data_sets[data_set],
    burn_in_fraction=burn_in_fraction,
    fasta_path=fasta_paths[data_set],
    phylo_model_specification=phylo_model_specification,
    branch_model_name=branch_model_name,
    scalar_model_name=scalar_model_name,
    optimizer_name=optimizer_name,
    particle_count=num_particles)

print(burro.estimate_elbo(num_particles))
```

Error message:
```
Assertion failed: (index >= 0 && index < size()), function operator[], file lib/eigen/Eigen/src/Core/DenseCoeffsBase.h, line 163.
Abort trap: 6
```

**Comment by junseonghwan** [comment]
With smaller number of particles, say `num_particles = 10` this doesn't problem doesn't seem to occur.

**Comment by junseonghwan** [comment]
```python
step_count = 100
burro.gradient_steps(step_count)
```

also triggers the problem.

**Comment by junseonghwan** [comment]
Among the topologies that are sampled, I see there is an index that is larger than the length of sbn_parameters_. The bug is probably due to having to split a subsplit that is not seen when unrooting .

**Comment by junseonghwan** [comment]
Fix implemented as part of #144.
